src/PopChainManager/PopChainManager.spec.ts

```diff
@@ -0,0 +1,64 @@
+/// <reference path='../../typings/index.d.ts' />
+
+import popChainManager from './';
+import popEngine from '../PopEngine';
+import Pop from '../Pop';
+import Trigger from '../Trigger';
+
+describe('PopChainManager - ', () => {
+
+  describe('maybePinOrUnpinPopAndParentPops() - ', () => {
+
+    it('should add pinned-pop attribute to elements in chain', () => {
+
+      let parentEl = document.createElement('div');
+      parentEl.setAttribute('popgun', '');
+      parentEl.setAttribute('popgun-group', 'parent');
+
+      let childEl = document.createElement('div');
+      childEl.setAttribute('popgun', '');
+      childEl.setAttribute('popgun-group', 'child');
+
+      let t = new Trigger('click');
+
+      let parentPop = new Pop(parentEl, t);
+      let childPop = new Pop(childEl, t);
+
+      popEngine.addPopToPopStore('parent', parentPop);
+      popEngine.addPopToPopStore('child', childPop);
+
+      popChainManager.setParentChildRelationship(parentPop, childPop);
+      popChainManager.maybePinOrUnpinPopAndParentPops(childPop.targetEl, true);
+
+      expect(childEl.hasAttribute('pinned-pop')).toBe(true);
+      expect(parentEl.hasAttribute('pinned-pop')).toBe(true);
+
+    });
+
+  });
+
+  describe('setParentChildRelationship() - ', () => {
+
+    it('should set parent-child relationship between pops', () => {
+
+      let parentEl = document.createElement('div');
+      parentEl.setAttribute('popgun', '');
+
+      let childEl = document.createElement('div');
+      childEl.setAttribute('popgun', '');
+
+      let t = new Trigger('click');
+
+      let parentPop = new Pop(parentEl, t);
+      let childPop = new Pop(childEl, t);
+
+      popChainManager.setParentChildRelationship(parentPop, childPop);
+
```

Comment:
Worth adding a simple check for parentPop.childPops.length === 1 and childPop.childPops.length === 0? Simple enough test for mutations I'd say, also potentially overkill.